src/main/java/org/logstash/beats/Message.java

```diff
@@ -0,0 +1,61 @@
+package org.logstash.beats;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Message implements Comparable<Message> {
+    private final int sequence;
+    private String identityStream;
+    private final Map data;
```

Comment:
can you keep this as `Map<String, String>` - prevents casting in the MessageEncoder